src/components/webcams/streamers/WebrtcGo2rtc.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+    <div>
+        <video
+            v-show="status === 'connected'"
+            ref="video"
+            :style="webcamStyle"
+            class="webcamImage"
+            autoplay
+            playsinline
+            muted />
+        <v-row v-if="status !== 'connected'">
+            <v-col class="_webcam_webrtc_output text-center d-flex flex-column justify-center align-center">
+                <v-progress-circular v-if="status === 'connecting'" indeterminate color="primary" class="mb-3" />
+                <span class="mt-3">{{ status }}</span>
+            </v-col>
+        </v-row>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Ref, Watch } from 'vue-property-decorator'
+import BaseMixin from '@/components/mixins/base'
+import { GuiWebcamStateWebcam } from '@/store/gui/webcams/types'
+import WebcamMixin from '@/components/mixins/webcam'
+
+@Component
+export default class WebrtcGo2rtc extends Mixins(BaseMixin, WebcamMixin) {
+    @Prop({ required: true }) readonly camSettings!: GuiWebcamStateWebcam
+    @Prop({ default: null }) readonly printerUrl!: string | null
+    @Ref() declare video: HTMLVideoElement
+
+    pc: RTCPeerConnection | null = null
+    ws: WebSocket | null = null
+    restartPause = 2000
+    restartTimeout: any = null
+    status: string = 'connecting'
+
+    mounted() {
+        this.start()
+    }
+
+    // stop the video and close the streams if the component is going to be destroyed so we don't leave hanging streams
+    beforeDestroy() {
+        this.terminate()
+
+        // clear any potentially open restart timeout
+        if (this.restartTimeout) clearTimeout(this.restartTimeout)
+    }
+
+    get webcamStyle() {
+        return {
+            transform: this.generateTransform(
+                this.camSettings.flip_horizontal ?? false,
+                this.camSettings.flip_vertical ?? false,
+                this.camSettings.rotation ?? 0
+            ),
+        }
+    }
+
+    get url() {
+        const urlSearch = new URL(this.camSettings.stream_url).search.toString()
```

Comment:
As I am reviewing a [PR in Fluidd](https://github.com/fluidd-core/fluidd/pull/1245) on this same type of stream source, I am struggling to see the point of this approach.

The whole objective of stream_url is that users supply the actual go2rtc stream URL, so in this case I think should be "ws://<host>:<port>/api/ws?src=<camera>&media=video+audio" so that this method would not be required.

Reply:
hey @pedrolamas! So the problem with go2rtc is that the stream/WebRTC URL is not visible/available in the web interface, but the URL of the WebRTC "viewer". So the user can use the "viewer url", and Mainsail converts it to the Websocket URL. If you find a better solution, I would be open to changing this so that we both use the same URL with this streamer.

The same "convert" will be also used in the go2rtc WebRTC viewer to convert the current URL (location.href) to the Websocket URL.

Reply:
Fair enough, that makes sense! 🙂